packages/text-field/index.js

```diff
@@ -0,0 +1,367 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import {MDCTextFieldFoundation} from '@material/textfield';
+
+import Input from './Input';
+import Icon from './icon';
+import HelperText from './helper-text';
+import FloatingLabel from '@material/react-floating-label';
+import LineRipple from '@material/react-line-ripple';
+import NotchedOutline from '@material/react-notched-outline';
+
+class TextField extends React.Component {
+
+  foundation_ = null;
+  getBadInput = () => {};
+  getIsValid = () => {};
+
+  constructor(props) {
+    super(props);
+    this.floatingLabelElement = React.createRef();
+    this.textFieldElement = React.createRef();
+
+    this.state = {
+      // line ripple state
+      activeLineRipple: false,
+
+      // root state
+      value: null,
+      classList: new Set(),
+      inputId: props.children.props.id,
+      isFocused: false,
+      isBadInput: false,
+      dir: 'ltr',
+      disabled: false,
+
+      // floating label state
+      shouldFloatLabel: false,
+      labelWidth: 0,
+
+      // line ripple state
+      lineRippleCenter: null,
+
+      // notched outline state
+      shouldNotchOutline: false,
+
+      // helper text state
+      helperTextShowToScreenReader: false,
+      isValid: true,
+    };
+  }
+
+  inputProps(props) {
+    const {foundationValue} = this.state;
+    return Object.assign({}, props, {
+      foundationValue,
+      foundation: this.foundation_,
+      updateFocus: (isFocused) => this.setState({isFocused}),
+      handleValueChange: (value) => this.setState({value}),
+      // These are callbacks for Input, which set validity.badInput &
+      // validity.valid.
+      setBadInputHandler: (getBadInput) => this.getBadInput = getBadInput,
+      setIsValidHandler: (getIsValid) => this.getIsValid = getIsValid,
+      setDisabled: (disabled) => this.setState({disabled}),
+      setInputId: (id) => this.setState({inputId: id}),
+    });
+  }
+
+  componentDidMount() {
+    const foundationMap = {
+      helperText: this.helperTextAdapter,
+    };
+    this.foundation_ = new MDCTextFieldFoundation(this.adapter, foundationMap);
+    this.foundation_.init();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.value !== prevState.value) {
+      this.foundation_.setValue(this.state.value);
+    }
+  }
+
+  componentWillUnmount() {
+    this.foundation_.destroy();
+  }
+
+  getIsRtl = () => {
+    if (this.textFieldElement.current) {
+      const dir = window.getComputedStyle(this.textFieldElement.current).getPropertyValue('direction');
+      return dir === 'rtl';
+    }
+  }
+
+  /**
+  * getters
+  */
+
+  get classes() {
```

Comment:
Nit: Move the getters above the constructor but below class field declarations to follow the MDCWeb structure. This would apply to all the new files in this PR.

Reply:
Spoke offline: this change would be inconsistent with other components. We will have a follow up change/PR to update all classes to follow same structure. When we do this, we should also implement some lint rules to enforce this going further.